handlers/addNewProduct.js

```diff
@@ -0,0 +1,84 @@
+const mysql = require('mysql')
+const { v4: uuidv4 } = require('uuid')
+require('dotenv').config()
+
+module.exports.createProduct = async (event, context) => {
+  console.log('Created database client!')
+  const client = mysql.createConnection({
+    host: process.env.AWS_RDS_ENDPOINT,
+    user: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_NAME,
+    port: process.env.DB_PORT,
+    connectTimeout: 10,
+  })
+
+  let response
+  try {
+    console.log('Connecting to database...')
+    await new Promise((resolve, reject) => {
+      client.connect(function (err) {
+        if (err) {
+          console.error(
+            'Something went wrong while connecting to the database. Retry again to connect it.'
+          )
+          reject(err.stack)
+        }
+        console.log('Connected as id ' + client.threadId)
+        resolve(client.threadId)
+      })
+    })
+
+    console.log('Inserting product...')
+    await new Promise((resolve, reject) => {
+      const { title, description, price, thumbnail, count } = JSON.parse(
+        event.body
+      )
+      const id = uuidv4()
+      client.query(

```

Comment:
Correct me if I am wrong but this construction doesn't looks like transactional.
It may lead to inconsistent data in database in case stock insertion would fail.